src/com/rpl/proxy_plus.clj

```diff
@@ -51,11 +51,48 @@
         (.getDeclaredMethods klass)
         ))))
 
-(defn- find-matching-method [^Class klass name num-params]
+(defn- tag-matches [tag param-class]
+  (or
+   (nil? tag)
+   (let [compare-class
+     (case tag
+       ;; map primitive type hints to their equivalent classes
+       byte Byte/TYPE
+       short Short/TYPE
+       int Integer/TYPE
+       long Long/TYPE
+       float Float/TYPE
+       double Double/TYPE
+       boolean Boolean/TYPE
+       char Character/TYPE
+       bytes (type (byte-array []))
+       shorts (type (short-array []))
+       ints (type (int-array []))
+       longs (type (long-array []))
+       floats (type (float-array []))
+       doubles (type (double-array []))
+       booleans (type (boolean-array []))
+       chars (type (char-array []))
+       (eval tag) ; default, just return the evaluated tag (which should be a Class)
```

Comment:
Why `eval`?

Reply:
At this point, the type hint is a symbol.  Evaluating that returns in the corresponding `java.lang.Class` for comparison to the Java reflection provided parameter classes.

Reply:
Can this callsite just use `resolve`? `eval` seems overkill.

Reply:
Yes, that turns out to be better.